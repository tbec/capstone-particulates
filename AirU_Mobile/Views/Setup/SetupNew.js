/**
 * Main setup page, navigates to login if not logged in otherwise navigates to first part of process (Review)
 */

import React, {Component} from 'react';
import {AsyncStorage} from 'react-native'
import Login from '../Setup/Login'
import { LOGIN_NAME } from '../../Components/Constants'
import {accountFuncs} from '../../Components/CommonFuncs'

export default class SetupNew extends Component<Props> {

    // checks if user has logged in previously
    constructor(Props) {
        super(Props);
        this.state = {loggedIn: false}
        this.checkLogin();
    }

    componentDidMount(props) {
        this.checkLogin();
    }

    /**
     * Checks if users has previously logged into system to display appropriate view
     */
    checkLogin() {
        accountFuncs.loginKeychain().then((_res) => {
            if (accountFuncs.loginKeychain() != null) {
                this.setState({ loggedIn: false})
            } else {
                this.setState({loggedIn: true})
                this.props.navigation.navigate('ReviewFirst')
            }
        })
        // AsyncStorage.getItem(LOGIN_NAME).then((_retrieveData) => {
        //     if (_retrieveData == null) {
        //         this.setState({ loggedIn: false})
        //     }
        //     else {
        //         this.setState({loggedIn: true})
        //         this.props.navigation.navigate('ReviewFirst');
        //     }
        // })
    }

    // If logged in already go to review via navigator, otherwise have login first
    render() {
        if (this.state.loggedIn) {            
            return null;
        }
        else {
            return (<Login navigation={this.props.navigation}/>)
        }
    }
}