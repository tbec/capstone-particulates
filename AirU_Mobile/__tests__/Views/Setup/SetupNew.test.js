import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import SetupNew from '../../../Views/Setup/SetupNew'
import MockStorage from '../../mockStorage'

// https://stackoverflow.com/questions/40952566/how-to-test-async-storage-with-jest
const storageCache = {};
const AsyncStorage = new MockStorage(storageCache);
jest.setMock('AsyncStorage', AsyncStorage)

// used to find testIDs in code
let findById = function(tree, testID) {
    if(tree.props && tree.props.testID === testID) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], testID)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}

test('Snapshot ConnectionSetup Setup', () => {
    const snap = renderer.create(<SetupNew/>).toJSON();
    expect(snap).toMatchSnapshot();
})

test ('Is Not Logged In', () => {
    let login = renderer.create(<SetupNew/>).getInstance();
    login.checkLogin();
    expect(login.state.loggedIn).toEqual(false)        
})

test ('Is Logged In', () => {
    let login = renderer.create(<SetupNew/>).getInstance();
    login.checkLogin();
    expect(login.state.loggedIn).toEqual(false)        
})