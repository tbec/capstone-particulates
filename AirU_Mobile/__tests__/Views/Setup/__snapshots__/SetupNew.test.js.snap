// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ConnectionSetup Setup 1`] = `
<RCTScrollView
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  getTextInputRefs={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onScroll={[Function]}
  scrollEventThrottle={200}
  scrollToBottomOnKBShow={false}
  scrollToInputAdditionalOffset={75}
  startScrolledToBottom={false}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#b3e6ff",
          "flex": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingTop": 20,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../Resources/red_cloud.jpeg",
            }
          }
          style={
            Object {
              "height": "60%",
              "width": "50%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "red",
              "fontWeight": "bold",
              "paddingBottom": 5,
            }
          }
        >
          Username
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          editable={true}
          keyboardType="default"
          onChangeText={[Function]}
          placeholder="Login"
          secureTextEntry={false}
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "black",
              "borderWidth": 1,
              "height": 40,
              "justifyContent": "center",
              "paddingBottom": 10,
              "width": "50%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "red",
              "fontWeight": "bold",
              "paddingBottom": 5,
            }
          }
        >
          Password
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          editable={true}
          keyboardType="default"
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "white",
              "borderColor": "black",
              "borderWidth": 1,
              "height": 40,
              "justifyContent": "center",
              "paddingBottom": 10,
              "width": "50%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <Text />
      <View
        accessibilityRole="button"
        accessibilityStates={
          Array [
            "disabled",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <Text
            disabled={true}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "blue",
                },
                Object {
                  "color": "#cdcdcd",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "red",
            "flex": 2,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "blue",
                },
              ]
            }
          >
            Register a new account
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
